extensions-core/druid-pac4j/src/main/java/org/apache/druid/security/pac4j/Pac4jFilter.java

```diff
@@ -96,24 +97,24 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
           "/",
           true, false, false, null);
     } else {
-      Object uid = securityLogic.perform(
+      OidcProfile profile = (OidcProfile) securityLogic.perform(
           context,
           pac4jConfig,
           (JEEContext ctx, Collection<UserProfile> profiles, Object... parameters) -> {
             if (profiles.isEmpty()) {
               LOGGER.warn("No profiles found after OIDC auth.");
               return null;
             } else {
-              return profiles.iterator().next().getId();
+              return profiles.iterator().next();
             }
           },
           JEEHttpActionAdapter.INSTANCE,
           null, "none", null, null);
       // Changed the Authorizer from null to "none".
       // In the older version, if it is null, it simply grant access and returns authorized.
       // But in the newer pac4j version, it uses CsrfAuthorizer as default, And because of this, It was returning 403 in API calls.
-      if (uid != null) {
-        AuthenticationResult authenticationResult = new AuthenticationResult(uid.toString(), authorizerName, name, null);
+      if (profile != null) {
```

Comment:
```suggestion
      if (profile != null && profile.get() != null) {
```
To preserve the old behaviour..